server/search/config/config.go

```diff
@@ -0,0 +1,17 @@
+// Copyright AGNTCY Contributors (https://github.com/agntcy)
+// SPDX-License-Identifier: Apache-2.0
+
+package config
+
+const (
+	DefaultDBType       = "sqlite"
+	DefaultSQLiteDBPath = "/data/search.db"
+)
+
+type Config struct {
+	// DBType is the type of the search database.
+	DBType string `json:"db_type,omitempty" mapstructure:"db_type"`
+
+	// SQLiteDBPath is the path to the SQLite database file.
+	SQLiteDBPath string `json:"sqlite_db_path,omitempty" mapstructure:"sqlite_db_path"`
```

Comment:
this can be registered here using provider-specific config struct, ie.

```go
// server/search/sqlite/config/config.go
type Config struct {
	DBPath string `json:"db_path,omitempty" mapstructure:"db_path"`
}

// server/search/config/config.go
type Config struct {
	// DBType is the type of the search database.
	DBType string `json:"db_type,omitempty" mapstructure:"db_type"`

	// Config for SQLite db provider
	SQLite sqliteconfig.Config `json:"sqlite,omitempty" mapstructure:"sqlite"`
}
```

Reply:
helps with extendability so that each provider can define their own configs independently